android/src/main/java/com/amplitude/android/migration/DatabaseStorage.kt

```diff
@@ -46,11 +46,13 @@ class DatabaseStorage(context: Context, databaseName: String) : SQLiteOpenHelper
     DatabaseConstants.DATABASE_VERSION
 ) {
     private var file: File = context.getDatabasePath(databaseName)
+    private var isValidDatabaseFile = true
     var currentDbVersion: Int = DatabaseConstants.DATABASE_VERSION
         private set
 
     override fun onCreate(db: SQLiteDatabase) {
-        throw NotImplementedError()
+        // File exists but it is not a legacy database for some reason.
```

Comment:
Curious why this could happen. Could this be a threading issue potentially solved here?
https://github.com/amplitude/Amplitude-Kotlin/pull/144/files

Reply:
I don't know - it is only a hypothesis.

> Could this be a threading issue potentially solved here?

The [PR 144](https://github.com/amplitude/Amplitude-Kotlin/pull/144) only covers file storage, the current bug is related to legacy sqlite storage.

Reply:
Can you please log an error here (by passing in a logger). We should provide a way for the user to know this issue occurred.

Reply:
Added logging.